### PR TITLE
feature/2.2/provide field usable operators

### PR DESCRIPTION
**Pull request by sensorario**
| Q             | A
| ------------- | ---
| Branch?       | 2.2
| Minor version?      | 2.2
| Hotfix?       | no  
| Refactoring?  | yes
| New feature?  | yes 
| BC breaks?    | yes/no
| Deprecations? | no 
| Tests pass?   | yes

This PR contains a little Dictionary refactoring. Dictionary's operators are now splitted in three families:

 - numbers
 - strings
 - fields

The component provide database map to the client with operator that can be used, for each field, to apply a filter. It is called Sherlock just because will helps in searches. It does not commit any search.

For each entity it provide a map with filters and relations

    {
        "fields": [],
        "relations": [],
    }

Each node will be like this:

    "appbundle.entity.entityname": {
        "fields": [
            "id": [">", "<", "=", ...],
            "username": ["contains", "endswith", ...],
        ],
        "relations": [
            "relName": "thebundle.entity.entityName"
        ],
    }

### CODE REVIEW

File: UPGRADE-2.2.md [hunk 1]
```diff
@@ -35,3 +35,31 @@ UPGRADE FROM 2.1 to 2.2
...
    manage the `foo|bar` part. It knows field name and operator name.
+
+Component\Sherlock
+------------------
+
+Added this new component. The component provide a service to obtain the entire
+database map with fields and operator available in each fields. Also, it
+provide for each entity the list of related relation.
+
+For example a field `id` (if id is a number) cannot be filtered with fitlers:
```
Comment (AlessandroMinoccheri) [inline UPGRADE-2.2.md]: typo in fitlers

File: src/Mado/QueryBundle/Component/Sherlock/CurrentMetaData.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    public function extractFields($entityClass) : array
+    {
+        $this->currentMetadata = $this->manager->getClassMetadata($entityClass);
+
+        return array_map(function ($item) {
+            return Dictionary::getOperatorsFromDoctrineType($item['type']);
+        }, $this->currentMetadata->fieldMappings);
+    }
+
+    public function haveRelations() : bool
```
Comment (AlessandroMinoccheri) [inline src/Mado/QueryBundle/Component/Sherlock/CurrentMetaData.php]: hasRelations maybe?

File: tests/Mado/QueryBundle/Component/Sherlock/SherlockTest.php [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                    ]
+                ]
+            ],
+            $this->sherlock->getOpList("mado.querybundle.tests.objects.startingentity")
+        );
+    }
+
+    public function testExtractJustFieldTypeAndNotOperatorList()
+    {
+        $this->assertEquals(
```
Comment (sensorario) [inline tests/Mado/QueryBundle/Component/Sherlock/SherlockTest.php]: @marcocasadio whenever will be necessary just field type there will be this method!

**Comment by AlessandroMinoccheri** [comment]
squash and merge

**Pull request merged**